Index: src/router/httpd/visuals/menu.c
===================================================================
--- src/router/httpd/visuals/menu.c	(revision 53494)
+++ src/router/httpd/visuals/menu.c	(working copy)
@@ -421,7 +421,7 @@
 				if (!strcmp_pnt(m->menu[i][j], "PPPoE_Server.asp"))
 					goto skip;
 #endif
-#ifdef HAVE_MICRO
+#if defined(HAVE_MICRO) || !defined(HAVE_PPTPD)
 				if (!strcmp_pnt(m->menu[i][j], "PPTP.asp"))	// jump over PPTP in
 					// micro build
 					goto skip;
@@ -444,7 +444,7 @@
 				if (!strcmp_pnt(m->menu[i][j], "Upgrade.asp"))
 					goto skip;
 #endif
-#ifdef HAVE_SANSFIL
+#if defined(HAVE_SANSFIL) || !defined(HAVE_HOTSPOT)
 				if (!strcmp_pnt(m->menu[i][j], "Hotspot.asp"))
 					goto skip;
 #endif
