Index: src/router/libutils/libutils/detect.c
===================================================================
--- src/router/libutils/libutils/detect.c	(revision 53494)
+++ src/router/libutils/libutils/detect.c	(working copy)
@@ -1101,7 +1101,7 @@
 			return ROUTER_DIR882;
 		}
 		if (!memcmp(sign, "DIR_878", 7)) {
-			setRouter("Dlink DIR-878 R1");
+			setRouter("Phicomm K2P");
 			return ROUTER_DIR882;
 		}
 	}
Index: src/router/libutils/libutils/gpio.c
===================================================================
--- src/router/libutils/libutils/gpio.c	(revision 53494)
+++ src/router/libutils/libutils/gpio.c	(working copy)
@@ -1116,6 +1116,25 @@
 
 void set_gpio(int pin, int value)
 {
+	if (pin == 13) {
+		writeint("/sys/class/leds/red:status/brightness", value);
+		writeint("/sys/class/leds/yellow:status/brightness", !value);
+		writeint("/sys/class/leds/blue:status/brightness", 0);
+		return;
+	}
+	if (pin == 14) {
+		writeint("/sys/class/leds/red:status/brightness", !value);
+		writeint("/sys/class/leds/yellow:status/brightness", value);
+		writeint("/sys/class/leds/blue:status/brightness", 0);
+		return;
+	}
+	if (pin == 15) {
+		writeint("/sys/class/leds/red:status/brightness", 0);
+		writeint("/sys/class/leds/yellow:status/brightness", !value);
+		writeint("/sys/class/leds/blue:status/brightness", value);
+		return;
+	}
+
 	if (pin > 95) {
 		set_linux_gpio(pin, value);
 	} else {
@@ -1126,6 +1145,9 @@
 
 int get_gpio(int pin)
 {
+	if (pin >= 13 && pin <= 15)
+		return 0;
+
 	int value;
 	if (pin > 95)
 		get_linux_gpio(pin);
Index: src/router/libutils/libutils/ledconfig.c
===================================================================
--- src/router/libutils/libutils/ledconfig.c	(revision 53494)
+++ src/router/libutils/libutils/ledconfig.c	(working copy)
@@ -598,12 +598,9 @@
 		cfg->usb_power = 0xa;
 		break;
 	case ROUTER_DIR882:
-		cfg->connected_gpio = GPIO_LOW | 0x3;
-		cfg->disconnected_gpio = GPIO_LOW | 0x4;
-		cfg->diag_gpio = GPIO_LOW | 0x8;
-		cfg->power_gpio = GPIO_LOW | 0x10;
-		cfg->usb_gpio = GPIO_LOW | 0xc;
-		cfg->usb_gpio1 = GPIO_LOW | 0xe;
+		cfg->connected_gpio = 0xf;
+		cfg->power_gpio = GPIO_LOW | 0xd;
+		cfg->wlan_gpio = 0xe;
 		break;
 	case ROUTER_DIR860LB1:
 		cfg->power_gpio = GPIO_LOW | 0xf;
Index: src/router/rc/resetbutton.c
===================================================================
--- src/router/rc/resetbutton.c	(revision 53494)
+++ src/router/rc/resetbutton.c	(working copy)
@@ -809,7 +809,7 @@
 static void handle_reset(void)
 {
 
-	if ((brand & 0x000f) != 0x000f) {
+	if ((brand & 0x000ff) != 0x000ff) {
 		dd_loginfo("resetbutton", "factory default.\n");
 #if !defined(HAVE_XSCALE) && !defined(HAVE_MAGICBOX) && !defined(HAVE_FONERA) && !defined(HAVE_WHRAG108) && !defined(HAVE_GATEWORX) && !defined(HAVE_LS2) && !defined(HAVE_CA8) && !defined(HAVE_TW6600) && !defined(HAVE_LS5) && !defined(HAVE_LSX) && !defined(HAVE_SOLO51)
 		led_control(LED_DIAG, LED_ON);
@@ -1124,7 +1124,7 @@
 		val = (get_gpio(7) << 7) | (get_gpio(18) << 18);
 		break;
 	case ROUTER_DIR882:
-		val = (get_gpio(7) << 7) | (get_gpio(15) << 15);
+		val = get_gpio(3) << 15;
 		break;
 	case ROUTER_R6800:
 		val = (get_gpio(18) << 18) | (get_gpio(12) << 12) | (get_gpio(14) << 14);
@@ -1477,9 +1477,6 @@
 	case ROUTER_TPLINK_ARCHERC3150:
 		sesgpio = 0x103;
 		break;
-	case ROUTER_DIR882:
-		sesgpio = 0x107;
-		break;
 	case ROUTER_R6800:
 		sesgpio = 0x112;
 		wifi24gpio = 0x10e;
