Index: src/router/rules/libevent.mk
===================================================================
--- src/router/rules/libevent.mk	(revision 53493)
+++ src/router/rules/libevent.mk	(working copy)
@@ -19,6 +19,7 @@
 	$(MAKE) -C libevent clean
 
 libevent-configure: openssl
+	cd libevent && autoreconf -fi
 	cd libevent && ./configure  --disable-debug-mode --prefix=/usr ac_cv_host=$(ARCH)-uclibc-linux --libdir=/usr/lib --target=$(ARCH)-linux --host=$(ARCH) CC="ccache $(ARCH)-linux-uclibc-gcc" \
 	CFLAGS="$(COPTS) $(MIPS16_OPT) -I$(TOP)/openssl/include -ffunction-sections -fdata-sections -Wl,--gc-sections" \
 	LDFLAGS="$(COPTS) $(MIPS16_OPT) -L$(TOP)/openssl" 
