Index: src/router/rules/nettle.mk
===================================================================
--- src/router/rules/nettle.mk	(revision 53457)
+++ src/router/rules/nettle.mk	(working copy)
@@ -24,8 +24,9 @@
 	RANLIB="$(ARCH)-linux-ranlib $(LTOPLUGIN)"
 
 nettle: openssl gmp pcre
+	make -C nettle clean
 	make -C nettle desdata CC_FOR_BUILD="gcc"
-	make -C nettle CC_FOR_BUILD="gcc"
+	make -C nettle CC_FOR_BUILD="gcc" -j4
 
 nettle-clean:
 	-make -C nettle clean
