Index: src/router/rules/smbd.mk
===================================================================
--- src/router/rules/smbd.mk	(revision 53457)
+++ src/router/rules/smbd.mk	(working copy)
@@ -9,7 +9,7 @@
 	    RANLIB="$(ARCH)-linux-ranlib $(LTOPLUGIN)"
 
 	cd smbd/tools-glib && ./autogen.sh
-	cd smbd/tools-glib && ./configure --prefix=/usr --host=$(ARCH)-linux --disable-shared --enable-static  --libdir=/usr/lib \
+	cd smbd/tools-glib && PKG_CONFIG_PATH=$(TOP)/_staging_static/usr/lib/pkgconfig ./configure --prefix=/usr --host=$(ARCH)-linux --disable-shared --enable-static  --libdir=/usr/lib \
 	    CFLAGS="$(COPTS) $(LTO) $(MIPS16_OPT) -D_GNU_SOURCE -DNEED_PRINTF -I$(TOP)/_staging_static/usr/include/glib-2.0 -I$(TOP)/_staging_static/usr/lib/glib-2.0/include -Drpl_malloc=malloc -ffunction-sections -fdata-sections -Wl,--gc-sections" \
 	    LDFLAGS="-L$(TOP)/_staging_static/usr/lib $(LDLTO) -ffunction-sections -fdata-sections -Wl,--gc-sections" \
 	    LIBNL_CFLAGS="-I$(TOP)/libnl/include" \
