Index: src/router/rules/util-linux.mk
===================================================================
--- src/router/rules/util-linux.mk	(revision 53457)
+++ src/router/rules/util-linux.mk	(working copy)
@@ -8,6 +8,7 @@
 	cd util-linux && autoheader
 	cd util-linux && autoreconf -vfi
 	cd util-linux && ./configure --host=$(ARCH)-linux-uclibc --prefix=/usr --libdir=/usr/tmp CFLAGS="$(COPTS) $(MIPS16_OPT) -fPIC -DNEED_PRINTF -I$(TOP)/ncurses/include" PKG_CONFIG="/tmp" \
+	--disable-makeinstall-chown \
 	--disable-rpath \
 	--enable-new-mount	\
 	--disable-tls		\
@@ -18,13 +19,14 @@
 	--without-ncursesw	\
 	--without-tinfo \
 	--without-ncurses
-	make -C util-linux
+	sed -i 's/link_all_deplibs=no/link_all_deplibs=yes/g' util-linux/libtool
+	make -C util-linux -j4
 
 util-linux-clean:
 	make -C util-linux clean
 
 util-linux: ncurses
-	make -C util-linux
+	make -C util-linux -j4
 
 util-linux-install:
 	make -C util-linux install DESTDIR=$(INSTALLDIR)/util-linux
