Index: src/router/services/sysinit/sysinit-rt2880.c
===================================================================
--- src/router/services/sysinit/sysinit-rt2880.c	(revision 53527)
+++ src/router/services/sysinit/sysinit-rt2880.c	(working copy)
@@ -152,34 +152,24 @@
 	case ROUTER_R6220:
 		insmod("thermal_sys");
 		insmod("hwmon");
-		insmod("compat");
-		insmod("mac80211");
-		if (!nvram_match("no_mt76", "1")) {
-			insmod("mt76");
-			insmod("mt76-connac-lib");
-			insmod("mt7615-common");
-			insmod("mt7615e");
-			insmod("mt76x02-lib");
-			insmod("mt76x2-common");
-			insmod("mt76x2e");
-			insmod("mt76x0-common");
-			insmod("mt76x0e");
-			insmod("mt7603e");
-		}
-		eval("swconfig", "dev", "eth0", "set", "reset", "1");
-		eval("swconfig", "dev", "eth0", "set", "enable_vlan", "1");
-		eval("swconfig", "dev", "eth0", "vlan", "1", "set", "ports", "0 1 2 3 6t");
-		eval("swconfig", "dev", "eth0", "vlan", "2", "set", "ports", "4 6t");
-		eval("swconfig", "dev", "eth0", "set", "apply");
-//              set_smp_affinity(20, 2);        // eth
-//              set_smp_affinity(22, 4);        //wifi1
-//              set_smp_affinity(23, 8);        // wifi2
-		nvram_seti("sw_cpuport", 6);
-		nvram_seti("sw_wan", 4);
-		nvram_seti("sw_lan1", 3);
-		nvram_seti("sw_lan2", 2);
-		nvram_seti("sw_lan3", 1);
-		nvram_seti("sw_lan4", 0);
+
+		eval("mtk_esw", "65", "0", "0");
+		eval("mtk_esw", "65", "16777261", "1");
+		eval("mtk_esw", "65", "0", "2");
+		eval("mtk_esw", "65", "0", "3");
+		eval("mtk_esw", "65", "0", "4");
+		eval("mtk_esw", "65", "0", "5");
+		eval("mtk_esw", "50", "0", "0");
+		eval("mtk_esw", "75", "0", "0");
+		eval("mtk_esw", "73", "10", "0");
+		eval("mtk_esw", "90", "0", "0");
+		eval("mtk_esw", "90", "0", "1");
+		eval("mtk_esw", "90", "0", "2");
+		eval("mtk_esw", "90", "0", "3");
+		eval("mtk_esw", "90", "0", "4");
+		eval("mtk_esw", "41", "63", "1");
+		eval("mtk_esw", "79", "0", "0");
+		eval("mtk_esw", "78", "0", "0");
 		break;
 	case ROUTER_R6800:
 		insmod("thermal_sys");
@@ -375,7 +365,8 @@
 #endif
 		break;
 	}
-	eval("ifconfig", "eth0", "up");
+	eval("ifconfig", "eth2", "up");
+	eval("ifconfig", "eth3", "up");
 
 	nvram_default_geti("port0vlans", 2);
 	nvram_default_geti("port1vlans", 1);
@@ -383,15 +374,21 @@
 	nvram_default_geti("port3vlans", 1);
 	nvram_default_geti("port4vlans", 1);
 	nvram_default_get("port5vlans", "1 2 16000");
+
 	eval("vconfig", "set_name_type", "VLAN_PLUS_VID_NO_PAD");
-	eval("vconfig", "add", "eth0", "1");	//LAN
-	eval("vconfig", "add", "eth0", "2");	//WAN
+	eval("vconfig", "add", "eth2", "1");	//LAN
+	eval("vconfig", "add", "eth3", "2");	//WAN
 
+	eval("ifconfig", "vlan1", "up");
+	eval("ifconfig", "vlan2", "up");
+
+	insmod("hw_nat");
+
 	struct ifreq ifr;
 	int s;
 
 	char eabuf[32];
-	if (get_hwaddr("eth0", eabuf)) {
+	if (get_hwaddr("eth2", eabuf)) {
 		nvram_set("et0macaddr_safe", eabuf);
 	}
 #else
